lib/postgres_ext/active_record/relation/query_methods.rb

```diff
@@ -1,70 +1,113 @@
 module ActiveRecord
   module QueryMethods
     class WhereChain
-      def overlap(opts)
-        opts.each do |key, value|
-          @scope = @scope.where(arel_table[key].overlap(value))
-        end
-        @scope
+      def overlap(opts, *rest)
+        substitute_comparisons(opts, rest, Arel::Nodes::Overlap, 'overlap')
       end
 
-      def contained_within(opts)
-        opts.each do |key, value|
-          @scope = @scope.where(arel_table[key].contained_within(value))
-        end
-
-        @scope
+      def contained_within(opts, *rest)
+        substitute_comparisons(opts, rest, Arel::Nodes::ContainedWithin, 'contained_within')
       end
 
-      def contained_within_or_equals(opts)
-        opts.each do |key, value|
-          @scope = @scope.where(arel_table[key].contained_within_or_equals(value))
-        end
-
-        @scope
+      def contained_within_or_equals(opts, *rest)
+        substitute_comparisons(opts, rest, Arel::Nodes::ContainedWithinEquals, 'contained_within_or_equals')
       end
 
-      def contains(opts)
-        opts.each do |key, value|
-          @scope = @scope.where(arel_table[key].contains(value))
+      def contains(opts, *rest)
```

Comment:
I'm not super happy with this bit, but it seems that the Arel maintainers want to do [less type casting](https://github.com/rails/arel/commit/008445d6fd5f825d9b445ac75a7be67f0f7ab52c) from inside Arel, so I moved it all into this method.